lib/modules/datasource/puppet-forge/index.ts

```diff
@@ -0,0 +1,62 @@
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import { PUPPET_FORGE } from './common';
+import type { PuppetModule } from './types';
+
+export class PuppetForgeDatasource extends Datasource {
+  static id = 'puppet-forge';
+
+  constructor() {
+    super(PuppetForgeDatasource.id);
+  }
+
+  override readonly defaultRegistryUrls = [PUPPET_FORGE];
+
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // https://forgeapi.puppet.com
+    const moduleSlug = packageName.replace('/', '-');
+    const url = `${registryUrl}/v3/modules/${moduleSlug}?exclude_fields=current_release`;
+
+    const { body: module } = await this.http.getJson<PuppetModule>(url);
+
+    const releases: Release[] = module.releases.map((release) => ({
+      version: release.version,
+      downloadUrl: release.file_uri,
+      releaseTimestamp: release.created_at,
+      registryUrl,
+    }));
+
+    if (!releases.length) {
+      return null;
+    }
+
+    return PuppetForgeDatasource.createReleaseResult(releases, module);
+  }
+
+  static createReleaseResult(
+    releases: Release[],
+    module: PuppetModule
+  ): ReleaseResult {
+    const result: ReleaseResult = {
+      releases,
+      homepage: module.homepage_url,
+      tags: {
+        moduleGroup: module.module_group,
+        premium: `${module.premium}`,
+      },
```

Comment:
Are these tags values used anywhere within Renovate? Today our only use is essentially npm, where you map tag->version and can follow a certain tag

Reply:
we are not using this tags actively. We thought it would be some kind of meta-data that could be useful. So should we remove it?

Reply:
I think best to remove it if unused, thanks